File: src/main/java/com/chaottic/nbt/MutableCompoundTag.java
package com.chaottic.nbt;

import com.chaottic.nbt.number.*;

import java.util.HashMap;

public final class MutableCompoundTag extends CompoundTag {

    public MutableCompoundTag() {
        super(new HashMap<>());
    }

    public void setByte(String name, byte b) {
        map.put(name, new ByteTag(b));
    }

    public void setShort(String name, short s) {
        map.put(name, new ShortTag(s));
    }

    public void setInt(String name, int i) {
        map.put(name, new IntTag(i));
    }

    public void setLong(String name, long l) {
        map.put(name, new LongTag(l));
    }

    public void setFloat(String name, float f) {
        map.put(name, new FloatTag(f));
    }

    public void setDouble(String name, double d) {
        map.put(name, new DoubleTag(d));
    }

    public void setString(String name, String s) {
        map.put(name, new StringTag(s));
    }

    public void setTag(String name, Tag tag) {
        map.put(name, tag);
    }
}

File: src/main/kotlin/com/chaottic/nbt/MutableCompoundTag.kt
package com.chaottic.nbt

import com.chaottic.nbt.number.*

class MutableCompoundTag private constructor(override val map: MutableMap<String, Tag>) : CompoundTag(map) {

	constructor() : this(hashMapOf())

	@JvmSynthetic
	operator fun set(name: String, b: Byte) {
		map[name] = ByteTag(b)
	}

	@JvmSynthetic
	operator fun set(name: String, s: Short) {
		map[name] = ShortTag(s)
	}

	@JvmSynthetic
	operator fun set(name: String, i: Int) {
		map[name] = IntTag(i)
	}

	@JvmSynthetic
	operator fun set(name: String, l: Long) {
		map[name] = LongTag(l)
	}

	@JvmSynthetic
	operator fun set(name: String, f: Float) {
		map[name] = FloatTag(f)
	}

	@JvmSynthetic
	operator fun set(name: String, d: Double) {
		map[name] = DoubleTag(d)
	}

	@JvmSynthetic
	operator fun set(name: String, s: String) {
		map[name] = StringTag(s)
	}

	@JvmSynthetic
	operator fun set(name: String, tag: Tag) {
		map[name] = tag
	}
}
